apm-lambda-extension/extension/process_events.go

```diff
@@ -18,30 +18,30 @@
 package extension
 
 import (
-	"context"
 	"encoding/json"
 	"net/http"
 )
 
 // ProcessShutdown processes the Shutdown event received from the
 // Lambda runtime API.
-func ProcessShutdown() {
+func ProcessShutdown(agentDataServer *http.Server, logsServer *http.Server) {
 	Log.Info("Received SHUTDOWN event, exiting")
 	agentDataServer.Close()
+	logsServer.Close()
 }
 
 // FlushAPMData reads all the apm data in the apm data channel and sends it to the APM server.
-func FlushAPMData(client *http.Client, dataChannel chan AgentData, config *extensionConfig, ctx context.Context) {
-	if !IsTransportStatusHealthyOrPending() {
-		Log.Debug("Flush skipped - Transport unhealthy")
+func FlushAPMData(transport *ApmServerTransport) {
```

Comment:
This function looks very similar to `StartBackgroundApmDataForwarding`. Probably a sign that it should be moved to `apm_server.go`.

Reply:
Thank you for this comment! It should definitely be - actually, I think we have a lot of room for improvement when it comes to the structure of the extension - see #192.

Reply:
Implemented in 0855915